src/components/Note.jsx

```diff
@@ -65,31 +66,31 @@ function Note(props) {
   /**
    * to pass the temp state to the main state
    */
-  const editSaveHandler = () => {
+  const editSaveHandler = useCallback(() => {
     mainStateDispatchFunction({
       type: 'edit',
       body: tempBody,
       color: tempBgColor,
       id: props.id,
     });
     setEditMode(false);
-  };
+  });
```

Comment:
ok here is the problem. you should add `setEditMode` to the dependency list.  Read more about DL